api/v1/ytsaurus_webhook.go

```diff
@@ -119,6 +120,35 @@ func (r *Ytsaurus) validateSecondaryMasters(old *runtime.Object) field.ErrorList
 	for i, sm := range r.Spec.SecondaryMasters {
 		path := field.NewPath("spec").Child("secondaryMasters").Index(i)
 		allErrors = append(allErrors, r.validateInstanceSpec(sm.InstanceSpec, path)...)
+		allErrors = append(allErrors, r.validateHostAddresses(r.Spec.PrimaryMasters, path)...)
+	}
+
+	return allErrors
+}
+
+func (r *Ytsaurus) validateHostAddresses(masterSpec MastersSpec, fieldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+
+	hostAddressesFieldPath := fieldPath.Child("hostAddresses")
+	if !r.Spec.HostNetwork && len(masterSpec.HostAddresses) != 0 {
+		allErrors = append(
+			allErrors,
+			field.Required(
+				field.NewPath("spec").Child("hostNetwork"),
+				fmt.Sprintf("%s doesn't make sense without hostNetwork=true", hostAddressesFieldPath.String()),
+			),
+		)
+	}
+
+	if len(masterSpec.HostAddresses) != 0 && len(masterSpec.HostAddresses) != int(masterSpec.InstanceCount) {
+		allErrors = append(
+			allErrors,
+			field.Invalid(
+				hostAddressesFieldPath,
+				masterSpec.HostAddresses,
+				fmt.Sprintf("%s list length shoud be equal to instanceCount", hostAddressesFieldPath.String()),
```

Comment:
Now I see that validation is a bit wrong. We should only check if any (or even only one of) of `HostAddresses` or `InstanceCount` is set, because `InstanceCount` makes no sense in that case and not used.